### DISCUSSION TITLE
uv development dependencies and tox

### DISCUSSION THREAD

**Discussion by astrojuanlu** [tox-dev/tox-uv#80] [Q&A] [answered]
When storing dev dependencies as optional dependencies, one could do

```
[testenv]
...
extras =
    test
commands =
    mypy src tests
    pytest {env:PYTEST_MARKERS:} {env:PYTEST_EXTRA_ARGS:} {posargs:-vv}
```

What would be the equivalent for [uv development dependencies](https://docs.astral.sh/uv/concepts/dependencies/#development-dependencies)?

**Comment by gaborbernat** [comment, marked as answer]
Currently the only way is optional dependency groups via extras. However, now via the `uv.lock` support one can use development dependencies too https://github.com/tox-dev/tox-uv?tab=readme-ov-file#uvlock-support

**Reply by astrojuanlu** [reply]
I see the `with_dev` flag. Thanks!
